src/BenchmarksApps/SslStream/Client/Program.cs

```diff
@@ -0,0 +1,225 @@
+
+        var task = Task.Run(async () =>
+        {
+            while (s_isRunning)
+            {
+                await EstablishConnection(options).ConfigureAwait(false);
+            }
+        });
+
+        await Task.Delay(options.Warmup).ConfigureAwait(false);
+        Log("Completing warmup...");
+
+        Stopwatch sw = Stopwatch.StartNew();
+        Volatile.Write(ref s_counters.TotalHandshakes, 0);
+
+        await Task.Delay(options.Duration).ConfigureAwait(false);
+        s_isRunning = false;
+        Log("Completing scenario...");
+
+        await task.ConfigureAwait(false);
+        sw.Stop();
+
+        LogMetric("sslstream/handshakes/mean", s_counters.TotalHandshakes / sw.Elapsed.TotalSeconds);
+    }
+
+    static async Task RunReadWriteScenario(ClientOptions options)
+    {
+        BenchmarksEventSource.Register("sslstream/read/mean", Operations.Avg, Operations.Avg, "Mean bytes read per second.", "Bytes per second - mean", "n0");
+        BenchmarksEventSource.Register("sslstream/write/mean", Operations.Avg, Operations.Avg, "Mean bytes written per second.", "Bytes per second - mean", "n0");
+
+        static async Task WritingTask(SslStream stream, int bufferSize, CancellationToken cancellationToken = default)
+        {
+            if (bufferSize == 0)
+            {
+                return;
+            }
+
+            var sendBuffer = new byte[bufferSize];
+            while (s_isRunning)
+            {
+                await stream.WriteAsync(sendBuffer, cancellationToken).ConfigureAwait(false);
+                Interlocked.Add(ref s_counters.BytesWritten, bufferSize);
+            }
+        }
+
+        static async Task ReadingTask(SslStream stream, int bufferSize, CancellationToken cancellationToken = default)
+        {
+            if (bufferSize == 0)
+            {
+                return;
+            }
+
+            var recvBuffer = new byte[bufferSize];
+            while (s_isRunning)
+            {
+                int bytesRead = await stream.ReadAsync(recvBuffer, cancellationToken).ConfigureAwait(false);
+                Interlocked.Add(ref s_counters.BytesRead, bytesRead);
+            }
+        }
+
+        using var sock = new Socket(SocketType.Stream, ProtocolType.Tcp);
+        await sock.ConnectAsync(options.Hostname, options.Port).ConfigureAwait(false);
+        using var stream = await EstablishSslStreamAsync(sock, options).ConfigureAwait(false);
+
+        byte[] recvBuffer = new byte[options.ReceiveBufferSize];
+
+        s_isRunning = true;
+
+        var writeTask = Task.Run(() => WritingTask(stream, options.SendBufferSize));
+        var readTask = Task.Run(() => ReadingTask(stream, options.ReceiveBufferSize));
+
+        await Task.Delay(options.Warmup).ConfigureAwait(false);
+        Log("Completing warmup...");
+
+        Stopwatch sw = Stopwatch.StartNew();
+        Volatile.Write(ref s_counters.BytesRead, 0);
+        Volatile.Write(ref s_counters.BytesWritten, 0);
```

Comment:
The way the two of them are not set together atomically (and also, most possibly, during some operation that currently is running) worries me a bit 😅

in HttpClient benchmarks, it's explicitly done [between iterations](https://github.com/aspnet/Benchmarks/blob/444b7bf3b253f85ef44e1efb89824fbdc5426352/src/BenchmarksApps/HttpClientBenchmarks/Clients/HttpClient/Program.cs#L303-L314), so I wonder whether we might want to do something like that here?